torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -347,6 +354,198 @@ at::Tensor tensor_from_cuda_array_interface(PyObject* obj) {
+    case at::kByte: memcpy_cpu<uint8_t> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((uint8_t *)tensor_ptr) + num_items);
+    case at::kChar: memcpy_cpu<char> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((char *)tensor_ptr) + num_items);
+    case at::kShort: memcpy_cpu<int16_t> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((int16_t *)tensor_ptr) + num_items);
+    case at::kInt: memcpy_cpu<int32_t> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((int32_t *)tensor_ptr) + num_items);
+    case at::kLong: memcpy_cpu<int64_t> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((uint64_t *)tensor_ptr) + num_items);
+    case at::kHalf: memcpy_cpu<at::Half> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((at::Half *)tensor_ptr) + num_items);
+    case at::kFloat: memcpy_cpu<float> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((float *)tensor_ptr) + num_items);
+    case at::kDouble: memcpy_cpu<double> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((double *)tensor_ptr) + num_items);
+    case at::kComplexHalf: memcpy_cpu<c10::complex<at::Half>> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((c10::complex<at::Half> *)tensor_ptr) + num_items);
+    case at::kComplexFloat: memcpy_cpu<c10::complex<float>> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((c10::complex<float> *)tensor_ptr) + num_items);
+    case at::kComplexDouble: memcpy_cpu<c10::complex<double>> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((c10::complex<double> *)tensor_ptr) + num_items);
+    case at::kBool: memcpy_cpu<bool> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((bool *)tensor_ptr) + num_items);
+    case at::kBFloat16: memcpy_cpu<at::BFloat16> (tensor_ptr, ndarray_ptr, num_items, itemsize, isaligned); return (void*)(((at::BFloat16 *)tensor_ptr) + num_items);
+    default: throw std::runtime_error("invalid type");
+  }
+  return NULL;
+}
+
+
+Tensor reverse_strides(Tensor & tensor) {
+  auto strides = tensor.strides();
+  auto size = strides.size();
+  std::vector<long int> new_strides(size);
+  new_strides[0] = strides[0];
+  for (int i=0; i<size-1; ++i) {
+    new_strides[i+1] = strides[size-i-1];
+  }
+  return at::native::set_storage_cpu_(tensor, tensor.storage(), tensor.storage_offset(), tensor.sizes(), new_strides);
+}
+
+
+at::Tensor tensor_from_ndarray_batch(std::vector<PyArrayObject*> & array_ptr_storage, std::vector<int64_t> & sizes, ScalarType scalarType, bool pin_memory) {
+  
+  size_t C_contiguous_count = 0;
+  bool C_contiguous_majority = true;
+
+  for (auto array:array_ptr_storage) {
+    if (PyArray_IS_C_CONTIGUOUS(array))
+      C_contiguous_count += 1;
+  }
+
+  if (2*C_contiguous_count < array_ptr_storage.size())
```

Comment:
curly braces here and on other if and for blocks, even if they only contain a single statement